DASH-MPD-PATCH-RESTRICTED.xsd

```diff
@@ -0,0 +1,71 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE schema [
+ <!ENTITY ncname "\i\c*">
+ <!ENTITY qname  "(&ncname;:)?&ncname;">
+ <!ENTITY aname  "@&qname;">
+ <!ENTITY pos    "\[\d+\]">
+ <!ENTITY attr   "\[&aname;='(.)*'\]|\[&aname;=&quot;(.)*&quot;\]">
+ <!ENTITY cond   "&attr;|&pos;">
+ <!ENTITY step   "(&qname;|\*)(&cond;)*">
+ <!ENTITY nspa   "namespace::&ncname;">
+ <!ENTITY last   "(&step;|&aname;|&nspa;)">
+]>
+<xs:schema xmlns:xs="http://www.w3.org/2001/XMLSchema" xmlns:patch="urn:ietf:params:xml:schema:patch-ops" xmlns="urn:mpeg:dash:schema:mpd-patch:2020" targetNamespace="urn:mpeg:dash:schema:mpd-patch:2020" elementFormDefault="qualified" attributeFormDefault="unqualified">
+    <!-- Import of patch operations from RFC5261 -->
+    <xs:import namespace="urn:ietf:params:xml:schema:patch-ops" schemaLocation="https://www.iana.org/assignments/xml-registry/schema/patch-ops.xsd"/>
+    <xs:element name="Patch" type="PatchType"/>
+    <!-- Patch -->
+    <xs:complexType name="PatchType">
+        <xs:choice minOccurs="0" maxOccurs="unbounded">
+            <xs:element name="add" type="add-restricted"/>
+            <xs:element name="remove" type="remove-restricted"/>
+            <xs:element name="replace" type="replace-restricted"/>
+            <xs:any namespace="##other" processContents="lax"/>
+        </xs:choice>
+        <xs:attribute name="mpdId" type="xs:string"/>
+        <xs:attribute name="publishTime" type="xs:dateTime"/>
+        <xs:attribute name="originalPublishTime" type="xs:dateTime"/>
+        <xs:anyAttribute namespace="##other" processContents="lax"/>
+    </xs:complexType>
+    <!-- Restricted XPath Selector for Replace/Remove Elements -->
+    <xs:simpleType name="xpath-restricted">
+        <xs:restriction base="patch:xpath">
+            <xs:pattern value="(/)?(&step;/)*(&last;)"/>
+        </xs:restriction>
+    </xs:simpleType>
+    <!-- Restricted XPath Selector for Add Elements -->
+    <xs:simpleType name="xpath-add-restricted">
+        <xs:restriction base="patch:xpath-add">
+            <xs:pattern value="(/)?(&step;/)*(&step;)"/>
```

Comment:
Similar to the above, simplified from previous add variant [`(/)?((&id;)((/&step;)*(/&child;))?|(&step;/)*(&child;))`](https://regexper.com/#%28%5C%2F%29%3F%28%28%26id%3B%29%28%28%5C%2F%26step%3B%29*%28%5C%2F%26child%3B%29%29%3F%7C%28%26step%3B%5C%2F%29*%28%26child%3B%29%29)

Same as before, `&id;` matcher was removed which eliminates the alternate pathways. The match remains:

- A possible root node indicator
- Followed by one or more steps, where step is a node with optional attribute or positional qualifiers

Unlike replace and remove add cannot end in a namespace or attribute selector. As above I pose the question on root node indicator optionality.